dags/glam_fog_release.py

```diff
@@ -0,0 +1,124 @@
+import operator
+from datetime import datetime, timedelta
+from functools import partial, reduce
+
+from airflow import DAG
+from airflow.operators.empty import EmptyOperator
+from airflow.sensors.external_task import ExternalTaskMarker, ExternalTaskSensor
+from airflow.utils.task_group import TaskGroup
+
+from operators.gcp_container_operator import GKEPodOperator
+from utils.constants import ALLOWED_STATES, FAILED_STATES
+from utils.glam_subdags.generate_query import (
+    generate_and_run_glean_task,
+)
+from utils.tags import Tag
+
+default_args = {
+    "owner": "efilho@mozilla.com",
+    "depends_on_past": False,
+    "start_date": datetime(2024, 12, 11),
+    "email": [
+        "telemetry-alerts@mozilla.com",
+        "akomarzewski@mozilla.com",
+        "efilho@mozilla.com",
+    ],
+    "email_on_failure": True,
+    "email_on_retry": True,
+    "retries": 1,
+    "retry_delay": timedelta(minutes=30),
+}
+
+PROJECT = "moz-fx-data-glam-prod-fca7"
+BUCKET = "moz-fx-data-glam-prod-fca7-etl-data"
+
+tags = [Tag.ImpactTier.tier_2]
+
+with DAG(
+    "glam_fog_release",
+    default_args=default_args,
+    max_active_runs=1,
+    schedule_interval="0 10 * * 6",  # 10am on Saturday
+    tags=tags,
+) as dag:
+    wait_for_glam_fog = ExternalTaskSensor(
+        task_id="wait_for_daily_glam_fog_release",
+        external_dag_id="glam_fog",
+        external_task_id="daily_release_done",
+        execution_delta=timedelta(hours=1),
```

Comment:
`glam_fog` runs at 2:00 so time delta should be 8 hours but also since this is weekly I think this needs an additional 1 week offset since the execution_date would be at the start of the interval, otherwise  it just waits for last week's task.  This is done in `firefox_public_data_report`:
https://github.com/mozilla/telemetry-airflow/blob/b8c132b9be38c636df23e66a11cae935fbfecaf5/dags/firefox_public_data_report.py#L45-L51

I think this is what it should be
```suggestion
        execution_delta=timedelta(days=-5, hours=-16),
```
or `timedelta(days=-6, hours=8)`.  I'm not sure though so it might be worth testing locally.

Reply:
Since this is weekly, the logical date of the dag would be a week before when it actually runs since it would be the start of the data interval.  So if it runs on 2024-12-14 with `timedelta=8 hours`, the logical date would be `2024-12-07 10:00` and it would wait for the task in `glam_fog` that has the date of `2024-12-07 02:00` which is the one that ran on `2024-12-08`.  -6 days would make it wait for the task that ran on 2024-12-14 since `glam_fog` is daily which means the logical date would be 1 day before when it runs.

Reply:
The [firefox_public_data_report](https://workflow.telemetry.mozilla.org/dags/firefox_public_data_report/grid?search=firefox_public_data_report&dag_run_id=scheduled__2024-12-02T01%3A00%3A00%2B00%3A00&tab=details&task_id=wait_for_main_ping) dag is an example of this.  It's weekly but waits for copy_deduplicate

Reply:
I see it now! The `logical_date` for weekly runs is at the start of the data interval. Thanks for taking the time to explain it twice 😬

Your suggestion below is correct, then
`execution_delta=timedelta(days=-5, hours=-16),`